ipv8/test/mocking/ipv8.py

```diff
@@ -48,9 +48,9 @@ def __init__(self, crypto_curve, overlay_class, create_trustchain=False, create_
             self.endpoint.enable_community_statistics(self.overlay.get_prefix(), True)
 
     def unload(self):
-        self.endpoint.close()
+        self.overlay.unload()
```

Comment:
What was the reason for swapping these statements? Since now it could be that there are incoming messages after unloading an overlay right?

Reply:
The reason was `None objects` kind of errors appearing in tests. At init, we first load the overlay, and then open the endpoint. Therefore, we *must* do it the other way around at shutdown, or the overlay tries to do stuff with the endpoint, which is not there...
The problem is, the overlay does not check if the endpoint object is still there.
For now its fine, no errors after this fix. But we'll have to clean up the init-shutdown sequences of overlays in the future.